lib/shelf_pubserver.dart

```diff
@@ -330,13 +330,20 @@ class ShelfPubServer {
           'message': 'Successfully uploaded package.',
         },
       });
-    } catch (error, stack) {
-      _logger.warning('An error occured while finishing upload', error, stack);
+    } on ClientSideProblem catch (error, stack) {
+      _logger.info('A problem occured while finishing upload.', error, stack);
       return _jsonResponse({
         'error': {
           'message': '$error.',
         },
       }, status: 400);
```

Comment:
I'd be nice if we were smarter about the 4xx status we sent. For instance, `UnauthorizedAcess`should be a 401, right?